### ISSUE TITLE
Which inference function is used in the experimental diagram of the paper

### ISSUE THREAD

**Issue by MoreanP** [suyoung-lee/LDM#3]
visualise_behaviour() and evaluate() are used in the log() function respectively
visualise_behaviour() function uses test_env, but the results are not written to the log and are only printed out
evaluate() function uses train_env, the result is written in the log
I plot the result diagram of the two functions and find that the result of evaluate() function is close to that of the paper, but the result of visualise_behaviour() function is poor
This is the result of three seeds in the environment of half_cheetah_vel
![image](https://user-images.githubusercontent.com/37109784/163918985-78595ed9-d9ef-4dc2-9195-b4c73e51d6f6.png)

**Comment by suyoung-lee** [comment]
We used visualise_behaviour function to report both the training and test scores in the paper. 
You are right that it does not write a log, but it stores the .npz files that contain "episode rewards".
evaluate() function reports the trainig score of randomly sampled trainig tasks But it is only used to keep track of training score . The result of this evaluate() is not used for any plot in the paper.

Please refer to utils/evaluation.py --> visualise_behaviour --> get_test_rollout 
np.savez(behaviour_dir + '_' + str(i) + '_data.npz',
                         episode_prev_obs=episode_prev_obs[i][:2].detach().cpu().numpy(),
                         episode_actions=episode_actions[i].detach().cpu().numpy(),
                         episode_rewards=episode_rewards[i].detach().cpu().numpy(),
                         )

It stores rewards for all tasks (training and test) under logs folder. 
e.g. logs/ldm_7__06:08_15:42:29/5500/0/00_1_data.npz
We aggregate the returns of test tasks and training tasks to report each scores respectively.
For the half cheetah vel environment,  00_...npz and 01_...npz files are used to report the trainig score and 02_...npz~06_...npz files are used to report the test score.

We use 8 random seeds from 0 to 7 for all experiments. 
I hope my answer helps you to reproduce the results.
